sql/src/main/java/org/apache/apex/malhar/sql/codegen/BeanClassGenerator.java

```diff
@@ -0,0 +1,796 @@
+
+          equalsNode.instructions.add(new JumpInsnNode(Opcodes.IFNONNULL, nonNullNode));
+          equalsNode.instructions.add(new JumpInsnNode(isLast ? Opcodes.IFNULL : Opcodes.IFNONNULL, isLast ? l4 : l3));
+
+          equalsNode.instructions.add(new JumpInsnNode(Opcodes.GOTO, continueNode));
+
+          equalsNode.instructions.add(nonNullNode);
+          equalsNode.instructions.add(new InsnNode(Opcodes.POP));
+          equalsNode.instructions.add(new VarInsnNode(Opcodes.ALOAD, 0));
+          equalsNode.instructions.add(new FieldInsnNode(Opcodes.GETFIELD, classNode.name, fieldName, fieldJavaType));
+          equalsNode.instructions.add(new VarInsnNode(Opcodes.ALOAD, 2));
+          equalsNode.instructions.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, classNode.name, getterMethodName,
+              "()" + fieldJavaType, false));
+          equalsNode.instructions.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, objectOwnerType, "equals",
+              "(Ljava/lang/Object;)Z", false));
+          equalsNode.instructions.add(new JumpInsnNode(isLast ? Opcodes.IFNE : Opcodes.IFEQ, isLast ? l4 : l3));
+
+          equalsNode.instructions.add(continueNode);
+          break;
+      }
+    }
+
+    equalsNode.instructions.add(l3);
+    equalsNode.instructions.add(new InsnNode(Opcodes.ICONST_0));
+    equalsNode.instructions.add(new InsnNode(Opcodes.IRETURN));
+
+    equalsNode.instructions.add(l4);
+    equalsNode.instructions.add(new InsnNode(Opcodes.ICONST_1));
+    equalsNode.instructions.add(new InsnNode(Opcodes.IRETURN));
+
+    classNode.methods.add(equalsNode);
+  }
+
+  private static String getJavaType(String fieldType)
+  {
+    Character pchar = PRIMITIVE_TYPES.get(fieldType);
+    if (pchar != null) {
+      //it is a primitive type
+      return Character.toString(pchar);
+    }
+    //non-primitive so find the internal name of the class.
+    return 'L' + fieldType.replace('.', '/') + ';';
+  }
+
+  /**
+   * Given the class name it reads and loads the class from the input stream.
+   *
+   * @param fqcn        fully qualified class name.
+   * @param inputStream stream from which class is read.
+   * @return loaded class
+   * @throws IOException
+   */
+  public static Class<?> readBeanClass(String fqcn, FSDataInputStream inputStream) throws IOException
+  {
+    byte[] bytes = IOUtils.toByteArray(inputStream);
+    inputStream.close();
+    return new ByteArrayClassLoader().defineClass(fqcn, bytes);
+  }
+
+  /**
+   * Given the class name it reads and loads the class from given byte array.
+   *
+   * @param fqcn       fully qualified class name.
+   * @param inputClass byte[] from which class is read.
+   * @return loaded class
+   * @throws IOException
+   */
+  public static Class<?> readBeanClass(String fqcn, byte[] inputClass) throws IOException
+  {
+    return new ByteArrayClassLoader().defineClass(fqcn, inputClass);
+  }
+
+  private static class ByteArrayClassLoader extends ClassLoader
```

Comment:
I didn't exactly created a top level class but rather a ClassLoaderUtils in lib which has this as public static class and also the method which uses this class.